base/show.jl

```diff
@@ -174,13 +174,31 @@ function show(io::IO, x::Core.IntrinsicFunction)
     print(io, unsafe_string(name))
 end
 
+show(io::IO, ::Core.BottomType) = print(io, "Union{}")
+
 function show(io::IO, x::Union)
     print(io, "Union")
-    sorted_types = sort!(collect(x.types); by=string)
+    sorted_types = sort!(uniontypes(x); by=string)
     show_comma_array(io, sorted_types, '{', '}')
 end
 
-show(io::IO, x::TypeConstructor) = show(io, x.body)
+function print_without_params(x::ANY)
+    if isa(x,UnionAll)
+        b = unwrap_unionall(x)
+        return isa(b,DataType) && b.name.wrapper === x
+    end
+    return false
+end
+
+function show(io::IO, x::UnionAll)
+    has_tvar_env = get(io, :tvar_env, false) !== false
+    if has_tvar_env && print_without_params(x)
+        return show(io, unwrap_unionall(x).name)
+    end
+    show(io, x.body)
+    print(io, " where ")
```

Comment:
It's a quantifier. e.g. `Vector{T} where T<:Integer` is the union of all types of the form `Vector{T}` where `T` is a subtype of `Integer`.

(Aside: I'll respond to these since they make a great FAQ, but of course will also add comments and docs later.)